cmd/istio-cni/nftables.go

```diff
@@ -0,0 +1,661 @@
+	if err != nil {
+		rdrct.logger.Errorf("Failed to get chains interface for table istio_ipv4 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	ipv6ci, err := ti.Tables().Table("istio_ipv6", definitions.TableFamilyIPv6)
+	if err != nil {
+		rdrct.logger.Errorf("Failed to get chains interface for table istio_ipv6 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	// Defining 2 base chains preruting and output
+	preRoutingAttr := nftableslib.ChainAttributes{
+		Type:     definitions.ChainTypeNAT,
+		Priority: 0,
+		Hook:     definitions.ChainHookPrerouting,
+	}
+	outputAttr := nftableslib.ChainAttributes{
+		Type:     definitions.ChainTypeNAT,
+		Priority: 0,
+		Hook:     definitions.ChainHookOutput,
+	}
+	// Creating ipv4 base chains for managing ipv4 related rules
+	if err := ipv4ci.Chains().CreateImm("istio_prerouting_ipv4", &preRoutingAttr); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_prerouting_ipv4 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	if err := ipv4ci.Chains().CreateImm("istio_output_ipv4", &outputAttr); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_output_ipv4 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	// Creating ipv6 base chains for managing ipv6 related rules
+	if err := ipv6ci.Chains().CreateImm("istio_prerouting_ipv6", &preRoutingAttr); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_prerouting_ipv6 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	if err := ipv6ci.Chains().CreateImm("istio_output_ipv6", &outputAttr); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_output_ipv6 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	// Creating ipv4 regularchains for managing ipv4 related rules
+	if err := ipv4ci.Chains().CreateImm("istio_inbound_ipv4", nil); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_inbound_ipv4 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	if err := ipv4ci.Chains().CreateImm("istio_outbound_ipv4", nil); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_outbound_ipv4 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	if err := ipv4ci.Chains().CreateImm("istio_redirect_ipv4", nil); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_redirect_ipv4 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	// Creating ipv6 regularchains for managing ipv6 related rules
+	if err := ipv6ci.Chains().CreateImm("istio_inbound_ipv6", nil); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_inbound_ipv6 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	if err := ipv6ci.Chains().CreateImm("istio_outbound_ipv6", nil); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_outbound_ipv6 with error: %+v\n", err)
+		return nil, nil, err
+	}
+	if err := ipv6ci.Chains().CreateImm("istio_redirect_ipv6", nil); err != nil {
+		rdrct.logger.Errorf("Failed to create istio_redirect_ipv6 with error: %+v\n", err)
+		return nil, nil, err
+	}
```

Comment:
This section looks like it could be significantly shorter by introducing a for-each loop.

Reply:
@luksa right, the reason I went this way was because it is new nftableslib api and I wanted it would be 120% clear what it does. I will make it a for loop and push the updated pr.

Reply:
@luksa please review latest revision.